### PR TITLE
Feature lwdev 7546 improve caching for assets asset pairs

### PR DESCRIPTION
**Pull request by njannink**

### CODE REVIEW

File: src/Lykke.Service.HFT.Core/Constants.cs [hunk 1]
```diff
@@ -1,7 +1,17 @@
-﻿namespace Lykke.Service.HFT.Core
+﻿using System;
+
+namespace Lykke.Service.HFT.Core
 {
     public static class Constants
     {
         public const string ComponentName = "HFT API (high-frequency-trading-service)";
+
+        public static ulong OrderCounter;
```
Comment (polonzer) [inline src/Lykke.Service.HFT.Core/Constants.cs]: This is not a production-like performance counters; they should not be in test / master branches.

File: src/Lykke.Service.HFT.Services/AssetServiceDecorator.cs [hunk 2]
```diff
@@ -1,25 +1,30 @@
...
+    public class AssetServiceDecorator : IAssetServiceDecorator, IDisposable
     {
         private readonly IAssetsServiceWithCache _apiService;
+        private IDisposable _updater;
 
         public AssetServiceDecorator(IAssetsServiceWithCache apiService)
         {
-            _apiService = apiService;
+            _apiService = apiService ?? throw new ArgumentNullException(nameof(apiService));
+            _updater = apiService.StartAutoCacheUpdate();
```
Comment (polonzer) [inline src/Lykke.Service.HFT.Services/AssetServiceDecorator.cs]: I suppose this API is not merged to test branch of AssetsService. We shouldn't use it right now.

**Comment by njannink** [comment]
Updated to latests assets nuget packageand removed dev performance counters

File: src/Lykke.Service.HFT.Core/Constants.cs [hunk 3]
```diff
@@ -1,4 +1,6 @@
-﻿namespace Lykke.Service.HFT.Core
+﻿using System;
```
Comment (polonzer) [inline src/Lykke.Service.HFT.Core/Constants.cs]: Why do we need this change?

File: src/Lykke.Service.HFT.Services/AssetServiceDecorator.cs [hunk 4]
```diff
@@ -1,25 +1,29 @@
...
+using System.Linq;
+using System.Threading.Tasks;
 
 namespace Lykke.Service.HFT.Services
 {
-    public class AssetServiceDecorator : IAssetServiceDecorator
+    public class AssetServiceDecorator : IAssetServiceDecorator, IDisposable
     {
         private readonly IAssetsServiceWithCache _apiService;
+        private IDisposable _updater;
```
Comment (polonzer) [inline src/Lykke.Service.HFT.Services/AssetServiceDecorator.cs]: Not used variable and IDIsposable.

File: src/Lykke.Service.HFT.Services/HealthService.cs [hunk 5]
```diff
@@ -1,6 +1,6 @@
-﻿using System.Collections.Generic;
-using Lykke.Service.HFT.Core.Domain.Health;
+﻿using Lykke.Service.HFT.Core.Domain.Health;
```
Comment (polonzer) [inline src/Lykke.Service.HFT.Services/HealthService.cs]: I suppose this file can be left as is.

File: src/Lykke.Service.HFT/Controllers/OrdersController.cs [hunk 6]
```diff
@@ -205,7 +206,8 @@ public class OrdersController : Controller
                 return BadRequest(ToResponseModel(ModelState));
             }
 
-            var assetPair = await _assetServiceDecorator.GetEnabledAssetPairAsync(order.AssetPairId);
+            var assetPair = await _assetServiceDecorator.GetAssetPairAsync(order.AssetPairId);
```
Comment (polonzer) [inline src/Lykke.Service.HFT/Controllers/OrdersController.cs]: GetEnabledAssetPairAsync

File: src/Lykke.Service.HFT.Services/AssetServiceDecorator.cs [hunk 7]
```diff
@@ -1,25 +1,29 @@
...
 
         public AssetServiceDecorator(IAssetsServiceWithCache apiService)
         {
-            _apiService = apiService;
+            _apiService = apiService ?? throw new ArgumentNullException(nameof(apiService));
         }
 
+        public async Task<AssetPair> GetAssetPairAsync(string assetPairId)
```
Comment (polonzer) [inline src/Lykke.Service.HFT.Services/AssetServiceDecorator.cs]: Why do we need new methods here?

File: src/Lykke.Service.HFT/Lykke.Service.HFT.csproj [hunk 8]
```diff
@@ -26,31 +26,22 @@
...
     <PackageReference Include="Lykke.SlackNotification.AzureQueue" Version="2.0.4" />
-    <PackageReference Include="Microsoft.ApplicationInsights.AspNetCore" Version="2.2.1" />
-    <PackageReference Include="Microsoft.AspNetCore" Version="2.0.2" />
-    <PackageReference Include="Microsoft.AspNetCore.Authentication" Version="2.0.3" />
-    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.0.3" />
-    <PackageReference Include="Microsoft.AspNetCore.StaticFiles" Version="2.0.2" />
-    <PackageReference Include="Microsoft.Extensions.Caching.Redis" Version="2.0.1" />
-    <PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="2.0.1" />
-    <PackageReference Include="Autofac.Extensions.DependencyInjection" Version="4.2.2" />
-    <PackageReference Include="Swashbuckle.AspNetCore" Version="2.4.0" />
```
Comment (polonzer) [inline src/Lykke.Service.HFT/Lykke.Service.HFT.csproj]: I would prefer to use this version of Swagger. We have already released HFT API with this version.

**Review by polonzer — COMMENTED** [review]
We need to wait until the related PR to AssetsService is approved.

File: src/Lykke.Service.HFT.Core/Constants.cs (hunk 3 above)
Comment (njannink) [reply src/Lykke.Service.HFT.Core/Constants.cs]: fixed

File: src/Lykke.Service.HFT.Services/AssetServiceDecorator.cs (hunk 4 above)
Comment (njannink) [reply src/Lykke.Service.HFT.Services/AssetServiceDecorator.cs]: removed

File: src/Lykke.Service.HFT.Services/HealthService.cs (hunk 5 above)
Comment (njannink) [reply src/Lykke.Service.HFT.Services/HealthService.cs]: reverted

File: src/Lykke.Service.HFT/Lykke.Service.HFT.csproj (hunk 8 above)
Comment (njannink) [reply src/Lykke.Service.HFT/Lykke.Service.HFT.csproj]: updated

File: src/Lykke.Service.HFT/Controllers/OrdersController.cs (hunk 6 above)
Comment (njannink) [reply src/Lykke.Service.HFT/Controllers/OrdersController.cs]: fixed

**Review by polonzer — APPROVED** [review]

**Pull request merged**